src/ui/chat-window.js

```diff
@@ -409,14 +415,17 @@ export class ChatScene extends Phaser.Scene {
             ) {
                 // arbitrary input length limit
                 if (promptInput.text.length < 15) {
-                    promptInput.text += event.key;
+                    promptInput.text =
+                        promptInput.text.substr(0, promptInput.text.length - 1) +
+                        event.key +
+                        "*";
                     promptInput.x -= 4;
                 }
             }
 
             // esc
             if (event.keyCode == 27) {
-                this.scene.destroyPrompt(promptObjs, keyboardInput, originallyVisible);
+                this.destroyPrompt(promptObjs, keyboardInput, originallyVisible);
```

Comment:
I think I'd prefer that escape closes chat rather than just removes the current prompt. This applies to both this and the previous one. For this one just one escape closes chat. The other prompt would need 1 to close prompt and another to close chat if it's still open.

Pressing enter or space to open the chat log would be convenient too